src/Pages/PrintingAnalytics/PrintingAnalytics.js

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useEffect } from 'react';
+import { getAllLogs } from '../../APIFunctions/2DPrinting';
+import './PrintingAnalytics.css';
+import { Line, Pie } from 'react-chartjs-2';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+export default function Analytics() {
+  const [printLogs, setPrintLogs] = useState([]);
+  const [filteredPrintLogs, setFilteredPrintLogs] = useState([]);
+  const [selectedDate, setSelectedDate] = useState(new Date());
+
+  useEffect(() => {
+    async function fetchPrintLogs() {
+      const fetchedLogs = await getAllLogs();
+      setPrintLogs(fetchedLogs.responseData);
+    }
+    fetchPrintLogs();
+  }, []);
+
+  // Filter the print logs to only get logs in selected date
+  useEffect(() => {
+    let targetDate = selectedDate
+      .toLocaleString('en-US', {hour12: false})
+      .split(',')[0];
+
+    let filteredPrintLogs = printLogs.filter(printLog => {
+      let dateAndTime = new Date(printLog.printedDate)
+        .toLocaleString('en-US', {hour12: false})
+        .split(',');
+
+      return dateAndTime[0] === targetDate;
+    });
+
+    setFilteredPrintLogs(filteredPrintLogs);
+  }, [selectedDate, printLogs]);
+
+  function getRequestCounts() {
+    let count = new Array(28);
+    for (let i = 0; i < count.length; i++) {
+      count[i] = 0;
+    }
+
+    for (let i = 0; i < filteredPrintLogs.length; i++) {
+      let dateAndTime = new Date(filteredPrintLogs[i].printedDate)
+        .toLocaleString('en-US', {hour12: false})
+        .split(',');
+      let time = dateAndTime[1].split(':');
+      let hour = parseInt(time[0].trim());
+      let minute = parseInt(time[1]);
+
+      // Only count requests made from [8, 22)
+      if (hour >= 8 && hour <= 21) {
+        if (minute < 30) {
+          count[(hour - 8) * 2] += 1;
+        } else {
+          count[((hour - 8) * 2) + 1] += 1;
+        }
+      }
+    }
+
+    return count;
+  }
+
+  function getPrinterCounts() {
+    let count = new Array(3);
+    for (let i = 0; i < count.length; i++) {
+      count[i] = 0;
+    }
+
+    for (let i = 0; i < filteredPrintLogs.length; i++) {
+      let chosenPrinter = filteredPrintLogs[i].chosenPrinter;
+      if (chosenPrinter === 'Fail') {
+        count[2] += 1;
+        continue;
+      }
+      chosenPrinter.split('-')[3] === 'left' ? count[0] += 1 : count[1] += 1;
+    }
+
+    return count;
+  }
+
+  const lineGraphInfo = {
+    labels: ['8:00', '8:30', '9:00', '9:30', '10:00', '10:30', '11:00',
```

Comment:
Evan and I wanted to make the labels more dynamic so that you can zoom out and in to show different times and dates, but there weren't any options on react-chartjs-2 that allowed that. If we wanted to change the labels so that more times are tracked, then the labels on the graph would be difficult to view. There weren't many options that we were able to do, so we just stuck with graphing from 8 to 21:30 in 30 min intervals.